### PR TITLE
[ODH-2285]: Airflow manager log parsing

### PR DESCRIPTION
**Pull request by EgorAfutin**
Log-parsing for airflow_manager components added.

### CODE REVIEW

File: src/applications/log_parsing/airflow/driver.py [hunk 1]
```diff
@@ -82,25 +108,66 @@ def create_event_creators(configuration):
...
-        ),
-        "/usr/local/airflow/logs": SourceConfiguration(
-            CompositeEventCreator()
-                .add_source_parser(general_creator)
+    clean_script_name_creator = MutateEventCreator(None, [
+        FieldsMapping(["script"], "script_name", lambda x: x.split(":")[0], True)
+    ])
+
+    message_level_rename_creator = MutateEventCreator(None, [
+        FieldsMapping(["level"], "message_level", lambda x: x, True)
```
Comment (halfwhite) [inline src/applications/log_parsing/airflow/driver.py]: lambda does nothing. Is it really effective? and what is True?

File: src/applications/log_parsing/airflow/driver.py [hunk 2]
```diff
@@ -40,6 +40,32 @@ def create_event_creators(configuration):
...
+    )
+
+    webui_manager_creator = EventCreator(
+        Metadata([
+            ConfigurableTimestampField("timestamp", timezone_name, timezones_property, "@timestamp"),
+            StringField("message"),
+            StringField("message_level")
+        ]),
+        RegexpParser(
+            r"^\[(?P<timestamp>\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2} \+\d*)\]\s+"
```
Comment (vunder-kind) [inline src/applications/log_parsing/airflow/driver.py]: Change "+' to any symbol.

File: src/applications/log_parsing/airflow/driver.py [hunk 3]
```diff
@@ -40,6 +40,32 @@ def create_event_creators(configuration):
...
+            r"(?P<message>.*\[(?P<message_level>[a-zA-Z]+)\]\s+.*)",
```
Comment (vunder-kind) [inline src/applications/log_parsing/airflow/driver.py]: Parse fields timestamp, thread_id, level, message.

File: src/applications/log_parsing/airflow/driver.py [hunk 4]
```diff
@@ -40,6 +40,32 @@ def create_event_creators(configuration):
         )
     )
 
+    general_manager_creator = EventCreator(
```
Comment (vunder-kind) [inline src/applications/log_parsing/airflow/driver.py]: There are no general parsing. Add 2 different event creators for scheduler_airflow and scheduler_latest logs.

File: src/applications/log_parsing/airflow/driver.py [hunk 5]
```diff
@@ -40,6 +40,32 @@ def create_event_creators(configuration):
...
+        Metadata([
+            ConfigurableTimestampField("timestamp", timezone_name, timezones_property, "@timestamp"),
+            StringField("script"),
+            StringField("level"),
+            StringField("message")
+        ]),
+        RegexpParser(
+            r"^\[(?P<timestamp>\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2},\d{3})\]\s+"
+            r"(?P<message>.*\{(?P<script>[^\}]+)\}.*\s+(?P<level>[a-zA-Z]+) - .*)"
```
Comment (vunder-kind) [inline src/applications/log_parsing/airflow/driver.py]: Use \w for words.

File: src/applications/log_parsing/airflow/driver.py [hunk 6]
```diff
@@ -82,25 +108,66 @@ def create_event_creators(configuration):
...
-            .add_intermediate_result_parser(crid_creator)
-            .add_intermediate_result_parser(clean_crid_creator, final=True)
-            .add_intermediate_result_parser(airflow_id_creator, final=True),
-            Utils.get_output_topic(configuration, 'worker')
-        ),
-        "/usr/local/airflow/logs": SourceConfiguration(
-            CompositeEventCreator()
-                .add_source_parser(general_creator)
+    clean_script_name_creator = MutateEventCreator(None, [
+        FieldsMapping(["script"], "script_name", lambda x: x.split(":")[0], remove_intermediate_fields=True)
```
Comment (vunder-kind) [inline src/applications/log_parsing/airflow/driver.py]: Why do you remove second part from script?

File: src/applications/log_parsing/airflow/driver.py [hunk 7]
```diff
@@ -82,25 +108,66 @@ def create_event_creators(configuration):
...
+    ])
+
+    message_level_rename_creator = MutateEventCreator(None, [
+        FieldsMapping(["level"], "message_level", simple_map, remove_intermediate_fields=True)
```
Comment (vunder-kind) [inline src/applications/log_parsing/airflow/driver.py]: What should this code do?

File: src/applications/log_parsing/airflow/driver.py [hunk 8]
```diff
@@ -82,25 +108,66 @@ def create_event_creators(configuration):
...
+    ])
+
+    airflow_manager_dag_creator = EventCreator(
+        Metadata([StringField("dag")]),
+        RegexpParser(r".*DAG?\(s\).*\['(?P<dag>.*)'\].*"),
```
Comment (vunder-kind) [inline src/applications/log_parsing/airflow/driver.py]: Remove "?" after DAG

Comment (vunder-kind) [reply src/applications/log_parsing/airflow/driver.py]: Use not greedy pattern for <dag>

File: src/applications/log_parsing/airflow/driver.py (hunk 5 above)
Comment (vunder-kind) [reply src/applications/log_parsing/airflow/driver.py]: Change pattern for script to .*?

File: src/applications/log_parsing/airflow/driver.py [hunk 9]
```diff
@@ -40,7 +40,79 @@ def create_event_creators(configuration):
         )
     )
 
-    dag_creator = EventCreator(
+    manager_scheduler_latest_event_creator = EventCreator(
```
Comment (halfwhite) [inline src/applications/log_parsing/airflow/driver.py]: Isn't it a too long method now?

File: src/common/log_parsing/dict_event_creator/mutate_event_creator.py [hunk 10]
```diff
@@ -50,13 +50,21 @@ def contains_fields_to_parse(self, row):
                           fields)).issubset(set(row.keys()))
 
 
+def concat_strings(x, y):
+    """
+    Concat two strings
+    """
+    # type: (str, str) -> str
```
Comment (halfwhite) [inline src/common/log_parsing/dict_event_creator/mutate_event_creator.py]: Take it inside doc comment.

File: src/test/unit/applications/log_parsing/airflow/test_airflow_log_parsing.py [hunk 11]
```diff
@@ -170,6 +179,138 @@ def test_airflow_with_subtask_and_airflow_id(self):
...
+
+    def test_airflow_manager_webui_without_script(self):
+        self.assert_parsing(
+            {
+                "topic": "airflowmanager_webui",
+                "source": "any.log",
+                "message": "[2018-04-12 10:12:16 +0000] [16262] [INFO] Booting worker with pid: 16262"
+            },
+            {
+                "@timestamp": datetime(2018, 4, 12, 10, 12, 16, 0).replace(tzinfo=timezones["Europe/Amsterdam"]),
```
Comment (vunder-kind) [inline src/test/unit/applications/log_parsing/airflow/test_airflow_log_parsing.py]: Wrong timestamp, use "idc" priority for driver

**Pull request merged**